flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/TaskMailboxImpl.java

```diff
@@ -251,4 +278,74 @@ public void quiesce() {
 	public State getState() {
 		return state;
 	}
+
+	@Override
+	public Mailbox getDownstreamMailbox(int priority) {
+		return new DownstreamMailbox(priority);
```

Comment:
Suggestion:
* make the `-1` constant (in `private DownstreamMailbox downstreamMailbox = new DownstreamMailbox(-1);`) a named `static final` field (for example `MIN_PRIORITY`);
* and add validation check here that `checkArgument(priority > MIN_PRIORITY)`.

Similarly, you can also add a `MAX_PRIORITY` constant (for example, it can be used for task cancellation letter) and additionally check here, that operator's priority is `< MAX_PRIORITY`.